lib/board.rb

```diff
@@ -0,0 +1,53 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize(cells = nil)
+    @cells = Array.new(9, " ")  
+  end
+
+  def reset!
+    initialize.clear
+    initialize
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    puts "\n"
+    puts "\n"
+  end
+
+  def position(position)
+    @cells[position.to_i-1]
+  end
+
+  def full?
+    @cells.all? do |char|
+      char == "X" || char == "O"
+    end
+  end
+
+  def turn_count
+    @cells.count do |count|
+      count == "X" || count == "O"
+    end
+  end
+
+  def taken?(char)
+    position(char) == "X" || position(char) == "O" ? true : false
```

Comment:
Avoid using the ternary operator to return `true` and `false`. Instead, rely on the implicit boolean return value of the expression. 

```
def taken?(char)
  position(char) == "X" || position(char) == "O"
end
```